### PR TITLE
Update Python format checker

### PR DESCRIPTION
**Pull request by coleramos425**
This PR:

- Updates [`formatting.yml`](https://github.com/ROCm/rocprofiler-compute/blob/colramos/update-format-check/.github/workflows/formatting.yml) workflow to run on the latest `3.x` Python version 
- Adds [`isort`](https://github.com/pycqa/isort) to Python formatting workflow to verify correctly organized import statements in Python files
- Adds an (optional) pre-commit hook for developers to catch any formatting errors before committing. Modifies `CONTRIBUTING.md` accordingly
- Fixes some YAML formatting errors that were caught using the new pre-commit hook
- Removes mention of deprecated "sign-off" requirement from `CONTRIBUTING.md`

Note that the "Formatting" check on this PR is currently failing. This is because `isort` has caught un-formatted import statements that require formatting. I'm leaving that re-formatting out of this PR to keep things neat. That re-formatting can happen in a separate PR.

### Proposed pre-commit hook usage
```console
python3 -m pip install pre-commit
cd rocprofiler-compute
pre-commit install
```
Now, moving forward any commits you propose to the repository will be automatically check via the pre-commit hook
image

### CODE REVIEW

**Comment by dgaliffiAMD** [comment]
Cool!
Maybe also add your setup instructions to the [CONTRIBUTING.md](https://github.com/ROCm/rocprofiler-compute/blob/amd-staging/CONTRIBUTING.md)?

File: pyproject.toml [hunk 1]
```diff
@@ -1,3 +1,14 @@
...
+[project.optional-dependencies]
+developer = [
+    "black>=22.6.0",
+    "isort>=5.12.0",
+    "pre-commit",
+]
+
 [tool.black]
 line-length = 90
 target-version = ['py38', 'py39']
```
Comment (cfallows-amd) [inline pyproject.toml]: I think it's already covered by requires-python at line 3 but should we target py310 and py311 as well here?

Comment (coleramos425) [reply pyproject.toml]: Good catch. I'll update

**Comment by dgaliffiAMD** [comment]
It looks like your changes are already catching a number of issues with the import sorting. Will you be patching them in this PR too? Or, will that be separate?

**Comment by coleramos425** [comment]
> It looks like your changes are already catching a number of issues with the import sorting. Will you be patching them in this PR too? Or, will that be separate?

What would you recommend? Initially, my thought was to leave those changes out to make the review process easier. But, if things look good and it would be easier, we can lob those changes into this PR too.

**Comment by dgaliffiAMD** [comment]
> > It looks like your changes are already catching a number of issues with the import sorting. Will you be patching them in this PR too? Or, will that be separate?
> 
> What would you recommend? Initially, my thought was to leave those changes out to make the review process easier. But, if things look good and it would be easier, we can lob those changes into this PR too.

Let's fix them, so we have a clean status check for your PR. 
It should be easy. I think  `python3 -m isort .` should catch and fix everything.

**Comment by coleramos425** [comment]
Done. Hook deployed. All checks now passing

image

**Review by dgaliffiAMD — APPROVED** [review]

**Review by cfallows-amd — APPROVED** [review]

**Pull request merged**